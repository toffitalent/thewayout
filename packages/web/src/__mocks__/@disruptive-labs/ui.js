const actual = jest.requireActual('@disruptive-labs/ui');

const showToast = jest.fn();

actual.Toast.show = showToast;
actual.Toast.danger = showToast;
actual.Toast.info = showToast;
actual.Toast.success = showToast;
actual.Toast.warning = showToast;

module.exports = {
  ...actual,
  showToast,
};
