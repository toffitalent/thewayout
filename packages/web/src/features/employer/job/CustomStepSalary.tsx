import {
  FieldComponentProps,
  FieldType,
  Flex,
  useWizardContext,
  WizardStep,
} from '@disruptive-labs/ui';
import { SalaryForm, SalaryWizardData } from './SalaryForm';

export interface CustomStepSalaryProps extends Omit<Partial<WizardStep>, 'fields'> {
  fields?: FieldComponentProps<FieldType>[];
}

export const CustomStepSalary = (props: CustomStepSalaryProps) => {
  const { next, data } = useWizardContext<SalaryWizardData>();

  const { fields } = props;
  if (!fields) {
    return null;
  }
  const [salaryField, minSalaryField, maxSalaryField, salaryDescriptionField] = fields;

  const handleSubmit = async (values: SalaryWizardData) => {
    await next(values);
  };

  return (
    <Flex maxWidth={96} ml="auto" mr="auto">
      <SalaryForm
        data={data}
        salaryField={salaryField}
        minSalaryField={minSalaryField}
        maxSalaryField={maxSalaryField}
        salaryDescriptionField={salaryDescriptionField}
        onSubmit={handleSubmit}
      />
    </Flex>
  );
};
