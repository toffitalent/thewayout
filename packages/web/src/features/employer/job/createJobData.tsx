import { Box, Text, WizardStep } from '@disruptive-labs/ui';
import {
  Experience,
  JobSalaries,
  offenseCategories,
  offenseText,
  states,
  TypeOfWork,
  VeteranOrJustice,
  WorkingTime,
} from '@two/shared';
import { CustomStepAllowedOffense } from '@app/components/CustomStepAllowedOffense';
import { InfoBox } from '@app/components/InfoBox';
import { job, JobEnums } from '@app/data/jobText';
import { months } from '@app/data/months';
import { WizardData } from './createJobDataTypes';
import { CustomStepDetails } from './CustomStepDetails';
import { CustomStepQuestions } from './CustomStepQuestions';
import { CustomStepResponsibilities } from './CustomStepResponsibility';
import { CustomStepSalary } from './CustomStepSalary';

export enum Steps {
  details = 'details',
  location = 'location',
  veteranOrJustice = 'veteranOrJustice',
  offenses = 'offenses',
  responsibilities = 'responsibilities',
  skills = 'skills',
  pay = 'pay',
  questions = 'questions',
}

export const stepsConfig: Record<Steps, { id: string; title: string }> = {
  [Steps.details]: { id: Steps.details, title: 'Add job details' },
  [Steps.location]: { id: Steps.location, title: 'Add job location' },
  [Steps.veteranOrJustice]: {
    id: Steps.veteranOrJustice,
    title: 'Who Can Apply For This Position?',
  },
  [Steps.offenses]: { id: Steps.offenses, title: 'Allowed Offenses' },
  [Steps.responsibilities]: {
    id: Steps.responsibilities,
    title: 'Responsibilities & Expectations',
  },
  [Steps.skills]: { id: Steps.skills, title: 'Skills & Experience' },
  [Steps.pay]: { id: Steps.pay, title: 'Pay & Benefits' },
  [Steps.questions]: { id: Steps.questions, title: 'Job Specific Questions' },
};

const years = Array.from({ length: 3 }, (_, i) => String(new Date().getFullYear() + i));

const getOptionsFromEnum = (enumObj: object) =>
  Object.entries(enumObj).map(([key, value]) => ({
    label: job[key as JobEnums],
    value,
  }));

export const steps: WizardStep[] = [
  {
    ...stepsConfig[Steps.details],
    description:
      'Please refrain from including company name or any employer-identifying information.',
    fields: [
      {
        label: 'Job Title',
        name: 'title',
        type: 'text',
        placeholder: 'E.g. Car Wash Attendant',
        validate: { required: true },
        fluid: true,
      },
      {
        label: 'Short Description',
        name: 'description',
        type: 'text',
        placeholder:
          'E.g. Wash attendants are responsible for performing the daily operational practices of running the car wash facility.',
        validate: { required: true, maxLength: 1000 },
        multiline: true,
        fluid: true,
        helperText: 'Max 1,000 characters.',
      },
      {
        label: 'Department',
        name: 'department',
        type: 'text',
        placeholder: 'E.g. Customer Service',
        validate: { required: true },
        fluid: true,
      },
      {
        name: 'startAtMonth',
        label: 'Start Date',
        type: 'select',
        options: Object.entries(months).map(([value, label]) => ({ label, value })),
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pr: 2,
        className: 'fieldHalfWidth',
        placeholder: 'Select month',
      },
      {
        name: 'startAtYear',
        type: 'select',
        options: years.map((el) => ({ label: el, value: el })),
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pl: 2,
        className: 'fieldHalfWidth',
        placeholder: 'Select year',
      },
      {
        label: 'Work Location',
        name: 'typeOfWork',
        type: 'select',
        options: getOptionsFromEnum(TypeOfWork),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: 'Employment Type',
        name: 'workingTime',
        type: 'select',
        options: getOptionsFromEnum(WorkingTime),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: 'Number of Open Positions',
        name: 'numberOfOpenPositions',
        placeholder: 'E.g. 3',
        type: 'number',
        validate: { required: true, min: 1 },
        fluid: true,
      },
    ],
    className: 'detailsStep',
    nextButtonLabel: 'Save & Next',
    component: <CustomStepDetails />,
    nextStep: (wizardData) =>
      (wizardData as WizardData).typeOfWork === TypeOfWork.onsite
        ? Steps.location
        : Steps.veteranOrJustice,
  },
  {
    ...stepsConfig[Steps.location],
    description: 'Where is the main office / place of work located?',
    fields: [
      {
        label: 'Job Address',
        name: 'address',
        type: 'text',
        placeholder: 'E.g. 7562 Center Ave',
        validate: { required: true },
      },
      {
        label: 'Town / City',
        name: 'city',
        type: 'text',
        placeholder: 'E.g. Los Angeles',
        validate: { required: true },
      },
      {
        label: 'State',
        name: 'state',
        type: 'select',
        options: Object.entries(states).map(([key, value]) => ({ value: key, label: value })),
        placeholder: 'E.g. California',
        validate: { required: true },
      },
      {
        label: 'Postcode / Zipcode',
        name: 'postalCode',
        type: 'text',
        placeholder: 'E.g. 92806',
        validate: { required: true },
      },
    ],
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.veteranOrJustice,
  },
  {
    ...stepsConfig[Steps.veteranOrJustice],
    description: 'Select all that apply.',
    fields: [
      {
        name: 'veteranOrJustice',
        type: 'choices',
        options: getOptionsFromEnum(VeteranOrJustice),
        multiple: true,
        inline: true,
        validate: (value) => (value.length >= 1 ? undefined : 'Error'),
      },
    ],
    showBackButton: true,
    nextButtonLabel: 'Save & Next',
    nextStep: (wizardData) =>
      (wizardData as WizardData).veteranOrJustice.includes(VeteranOrJustice.justiceImpacted)
        ? Steps.offenses
        : Steps.responsibilities,
  },
  {
    ...stepsConfig[Steps.offenses],
    description:
      'Only candidates that meet your allowable offenses (per State and Federal statutes) can apply for the job listing & will have their profile shared with you.',
    fields: Object.entries(offenseCategories).map(([key, values]) => ({
      name: `offense${key[0].toUpperCase() + key.slice(1)}`,
      label: values.name,
      type: 'choices',
      options: values.categories.map((el) => ({ label: offenseText[el], value: el })),
      inline: true,
      multiple: true,
      className: 'offenseCategoryStep',
    })),
    component: <CustomStepAllowedOffense />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.responsibilities,
  },
  {
    ...stepsConfig[Steps.responsibilities],
    description: 'Add your job responsibilities one at a time. Be mindful of inclusive language.',
    component: <CustomStepResponsibilities />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.skills,
  },
  {
    ...stepsConfig[Steps.skills],
    fields: [
      {
        label: 'Experience',
        name: 'experience',
        type: 'select',
        options: getOptionsFromEnum(Experience),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: (
          <Box display="inline-flex" alignItems="center">
            <Text fontWeight="700" mr={2} mb={0}>
              Skills - Description
            </Text>
            <InfoBox content="Explain the level of experience someone needs to be successful in this job." />
          </Box>
        ),
        name: 'skillsDescription',
        type: 'text',
        multiline: true,
        validate: { required: true, maxLength: 1000 },
        placeholder:
          "E.g. The most successful and happy people have PMA â€“ a Positive Mental Attitude! We are a huge believer in putting attitude above aptitude. We can always teach you how to do the job, but we can't teach a Positive Mental Attitude... You either have it or you don't!",
        helperText: 'Max 1,000 characters.',
      },
    ],
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.pay,
  },
  {
    ...stepsConfig[Steps.pay],
    fields: [
      {
        label: 'Salary',
        name: 'salary',
        type: 'select',
        options: getOptionsFromEnum(JobSalaries),
        validate: { required: true },
        placeholder: 'Select',
      },
      {
        label: 'Min Salary',
        name: 'min',
        type: 'text',
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pr: 2,
        className: 'fieldHalfWidth',
      },
      {
        label: 'Max Salary',
        name: 'max',
        type: 'text',
        validate: { required: true },
        width: '1/2',
        display: 'inline-block',
        pl: 2,
        className: 'fieldHalfWidth',
      },
      {
        label: (
          <Box display="inline-flex" alignItems="center">
            <Text fontWeight="700" mr={2} mb={0}>
              Pay Description
            </Text>
            <InfoBox content="Can include frequency of pay, when applicants can expect their first pay check, and any other helpful pay information" />
          </Box>
        ),
        name: 'salaryDescription',
        type: 'text',
        multiline: true,
        validate: { required: true, maxLength: 1000 },
        placeholder:
          'E.g. Starting wage is $13 per hour + tips, with an Employer Bonus after positive a 90 Day Review (provided by the employer)',
        fluid: true,
        helperText: 'Max 1,000 characters.',
      },
    ],
    component: <CustomStepSalary />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: Steps.questions,
  },
  {
    ...stepsConfig[Steps.questions],
    description: 'Ask qualifying questions that will help identify the most fit candidates.',
    component: <CustomStepQuestions />,
    nextButtonLabel: 'Save & Next',
    showBackButton: true,
    nextStep: 'confirm',
  },
];
