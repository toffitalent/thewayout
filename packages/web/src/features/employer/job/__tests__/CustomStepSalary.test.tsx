import { useWizardContext } from '@disruptive-labs/ui';
import { render } from '@test';
import { Steps, steps } from '../createJobData';
import { CustomStepSalary, CustomStepSalaryProps } from '../CustomStepSalary';

jest.mock('@disruptive-labs/ui', () => ({
  ...jest.requireActual('@disruptive-labs/ui'),
  useWizardContext: jest.fn(),
}));

describe('CustomStepSalary', () => {
  beforeEach(() => {
    (useWizardContext as any).mockImplementation(() => ({
      data: {},
      setData: jest.fn(),
      goTo: jest.fn(),
    }));
  });

  test('renders correctly', async () => {
    const salaryStep = steps.find((el) => el.id === Steps.pay);

    const { container, findByText } = render(
      <CustomStepSalary fields={(salaryStep as CustomStepSalaryProps)?.fields || []} />,
    );
    expect(await findByText('Salary')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
