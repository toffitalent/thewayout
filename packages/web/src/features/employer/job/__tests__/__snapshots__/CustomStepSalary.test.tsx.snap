// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStepSalary renders correctly 1`] = `
<div>
  <div
    class="maxWidth-96 marginRight-auto marginLeft-auto"
  >
    <form>
      <div
        class="control"
        role="group"
      >
        <label
          class="label"
          for=":r0:"
        >
          Salary
        </label>
        <div
          class="root primary md fluid md"
        >
          <select
            class="input"
            id=":r0:"
            name="salary"
          >
            <option
              disabled=""
              hidden=""
              value=""
            >
              Select
            </option>
            <option
              value="yearly"
            >
              Yearly salary
            </option>
            <option
              value="hourly"
            >
              Hourly salary
            </option>
          </select>
          <div
            class="icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="control display-inline-block width-1/2 paddingRight-2"
        role="group"
      >
        <label
          class="label"
          for=":r1:"
        >
          Min Salary
        </label>
        <div
          class="root primary md md fieldHalfWidth"
        >
          <input
            class="input"
            id=":r1:"
            name="min"
            placeholder="Min salary"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="control display-inline-block width-1/2 paddingLeft-2"
        role="group"
      >
        <label
          class="label"
          for=":r2:"
        >
          Max Salary
        </label>
        <div
          class="root primary md md fieldHalfWidth"
        >
          <input
            class="input"
            id=":r2:"
            name="max"
            placeholder="Max  salary"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="control"
        role="group"
      >
        <label
          class="label"
          for=":r3:"
        >
          <div
            class="alignItems-center display-inline-flex"
          >
            <p
              class="text marginRight-2 marginBottom-0 fontWeight-700"
            >
              Pay Description
            </p>
            <svg
              aria-controls="popover-:r4:"
              aria-expanded="false"
              aria-haspopup="dialog"
              aria-hidden="true"
              class="icon color-grey-500 width-7 height-7"
              focusable="false"
              id="popover-:r4:-trigger"
              stroke-width="2"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                stroke="none"
              />
              <path
                d="M3 12a9 9 0 1018 0 9 9 0 00-18 0M12 9h.01"
              />
              <path
                d="M11 12h1v4h1"
              />
            </svg>
            <div
              class="popover"
              data-popper-escaped="true"
              data-popper-placement="top"
              data-popper-reference-hidden="true"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; min-width: max-content; will-change: transform, visibility; bottom: 0px; transform: translate(0px, -8px);"
            >
              <div
                class="content popoverWrapper rounded-xl"
                id="popover-:r4:"
                role="dialog"
                tabindex="-1"
              >
                <div
                  class="arrow popoverArrow"
                  data-popper-arrow=""
                  style="will-change: transform; position: absolute; left: 0px; transform: translate(0px, 0px);"
                />
                <div
                  class="rounded-xl bgcolor-grey-100 maxWidth-96 paddingTop-5 paddingRight-5 paddingBottom-5 paddingLeft-5"
                >
                  Can include frequency of pay, when applicants can expect their first pay check, and any other helpful pay information
                </div>
              </div>
            </div>
          </div>
        </label>
        <div
          class="root primary md fluid md multiline"
        >
          <textarea
            aria-describedby=":r3:-helptext"
            class="input"
            id=":r3:"
            name="salaryDescription"
            placeholder="E.g. Starting wage is $13 per hour + tips, with an Employer Bonus after positive a 90 Day Review (provided by the employer)"
            rows="4"
            type="text"
          />
        </div>
        <div
          aria-live="polite"
          class="helperText"
          id=":r3:-helptext"
        >
          Max 1,000 characters.
        </div>
      </div>
      <div
        class="marginTop-4 textAlign-left"
      >
        <p
          class="text marginBottom-2 fontWeight-700"
        >
          Add Bonus
        </p>
        <div
          class="marginBottom-3"
        >
          <div
            class="control"
            role="group"
          >
            <div
              class="root primary md fluid md multiline"
            >
              <textarea
                aria-describedby=":r5:-helptext"
                class="input"
                id=":r5:"
                name="bonuses.0.value"
                placeholder="Additional $100 Bonus for Positive 90 Day Review"
                rows="4"
                type="text"
              />
            </div>
            <div
              aria-live="polite"
              class="helperText"
              id=":r5:-helptext"
            >
              Max 1,000 characters.
            </div>
          </div>
        </div>
        <button
          class="button md outline c-primary disabled marginTop-5"
          disabled=""
          type="button"
        >
          Add more
        </button>
      </div>
      <button
        class="button md solid c-primary disabled fluid maxWidth-96 marginTop-8"
        disabled=""
        type="submit"
      >
        Save & Next
      </button>
    </form>
  </div>
</div>
`;
